compiler/src/dotty/tools/dotc/transform/PatternMatcher.scala

```diff
@@ -834,7 +834,9 @@ object PatternMatcher {
       }
       plan match {
         case plan: TestPlan =>
-          val switchCases = collectSwitchCases(plan)
+          val switchCases =
+            if (plan.scrutinee.tpe.widen.classSymbol == defn.AnyClass) Nil
+            else collectSwitchCases(plan)
```

Comment:
I think it should be part of `collectSwitchCases` and use the already defined `isSwitchableType` function